runtime/src/accounts_db.rs

```diff
@@ -4913,6 +4933,33 @@ impl AccountsDb {
         self.do_load(ancestors, pubkey, None, load_hint, LoadZeroLamports::None)
     }
 
+    pub fn account_matches_owners(
+        &self,
+        ancestors: &Ancestors,
+        account: &Pubkey,
+        owners: &[&Pubkey],
+    ) -> Option<bool> {
+        let (slot, storage_location, _maybe_account_accesor) =
+            self.read_index_for_accessor_or_load_slow(ancestors, account, None, false)?;
+
+        if !storage_location.is_cached() {
+            let result = self.read_only_accounts_cache.load(*account, slot);
+            if let Some(account) = result {
+                return Some(owners.contains(&account.owner()));
+            }
+        }
+
```

Comment:
It seems like this section of code was repurposed from here: https://github.com/solana-labs/solana/blob/c7fb4b10401a10c21f27a2ccf47176ccb659b646/runtime/src/accounts_db.rs#L5341-L5357. 

@jeffwashington  is it important to also make the zero lamports check https://github.com/solana-labs/solana/blob/c7fb4b10401a10c21f27a2ccf47176ccb659b646/runtime/src/accounts_db.rs#L5350C5-L5351 in order to be consistent? I.e. right now when we look up accounts with zero lamports it returns `None`, what should be done here in those cases?

Reply:
I tested with a 0 lamport account, and the owner gets set to `SystemProgram`. I don't think the check is useful for our usecase (filtering executables/programs). If some other usecase tries to filter `SystemProgram` accounts, 0 lamport account will show up there. Not sure if that's a problem though.

Reply:
I think the answer will be wrong if the account is zero lamport and you did look for system program owner. The result should be 'false' for zero lamport account to be consistent.

Reply:
Sorry, missed this comment. I'll push another update.